noaa.go

```diff
@@ -33,90 +33,147 @@ type PointsResponse struct {
+	ValidTime string              `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
+	Value     []*WeatherValueItem `json:"value"`
+}
+
+// Weather holds the JSON value for the weather object.
+type Weather struct {
+	Values []*WeatherValue `json:"values"`
+}
+
+// HazardValueItem holds a value item from a GridpointForecastResponse's hazard.values[x].value[x].
+type HazardValueItem struct {
+	Phenomenon   string `json:"phenomenon"`
+	Significance string `json:"significance"`
+	EventNumber  int32  `json:"event_number"`
+}
+
+// HazardValue holds a hazard value from a GridpointForecastResponse's hazard.values[x].
+type HazardValue struct {
+	ValidTime string             `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
+	Value     []*HazardValueItem `json:"value"`
+}
+
+// Hazard holds a hazard item from a GridpointForecastRespones's hazards
+type Hazard struct {
+	Values []*HazardValue `json:"values"`
+}
+
+type HourlyForecastResponse struct {
+	Updated           string                          `json:"updated"`
+	Units             string                          `json:"units"`
+	ForecastGenerator string                          `json:"forecastGenerator"`
+	GeneratedAt       string                          `json:"generatedAt"`
+	UpdateTime        string                          `json:"updateTime"`
+	ValidTimes        string                          `json:"validTimes"`
+	Periods           []*ForecastResponsePeriodHourly `json:"periods"`
+	Point             *PointsResponse
 }
 
 // GridpointForecastResponse holds the JSON values from /gridpoints/<cwa>/<x,y>"
 // See https://weather-gov.github.io/api/gridpoints for information.
 type GridpointForecastResponse struct {
 	// capture data from the forecast
-	Updated   string `json:"updateTime"`
-	Elevation struct {
-		Value float64 `json:"value"`
-		Units string  `json:"unitCode"`
-	} `json:"elevation"`
-	Weather struct {
-		Values []struct {
-			ValidTime string `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
-			Value     []struct {
-				Coverage  string `json:"coverage"`
-				Weather   string `json:"weather"`
-				Intensity string `json:"intensity"`
-			} `json:"value"`
-		} `json:"values"`
-	} `json:"weather"`
-	Hazards struct {
-		Values []struct {
-			ValidTime string `json:"validTime"` // ISO 8601 time interval, e.g. 2019-07-04T18:00:00+00:00/PT3H
-			Value     []struct {
-				Phenomenon   string `json:"phenomenon"`
-				Significance string `json:"significance"`
-				EventNumber  int32  `json:"event_number"`
-			} `json:"value"`
-		} `json:"values"`
-	} `json:"hazards"`
+	Updated                          string                      `json:"updateTime"`
+	Elevation                        *ForecastElevation          `json:"elevation"`
+	Weather                          *Weather                    `json:"weather"`
+	Hazards                          *Hazard                     `json:"hazards"`
```

Comment:
Same for these, if they can be non-pointer types, we should push for that.

Reply:
Ack.